### PR TITLE
Disallow reading Delta Lake tables if column mapping is not none

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Disallow reading `delta.columnMapping.mode='name'` tables
Relates to #12592

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
# Delta Lake
* Disallow reading Delta Lake tables if `delta.columnMapping.mode` table property is not none. ({issue}`12621`)
```

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java [hunk 1]
```diff
@@ -150,6 +157,10 @@ public ConnectorPageSource createPageSource(
                     updateResultJsonCodec);
         }
 
+        if (getColumnMappingMode(table.getMetadataEntry()) == ColumnMappingMode.NAME) {
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java:160]: `getColumnMappingMode(table.getMetadataEntry()) != ColumnMappingMode.NONE` might be better.

[the remark this answers is not in the document]
Comment (alexjo2144) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java:160]: Yeah, I think we should be checking the minReaderVersion from the protocol definition instead. It'd also be nice to catch this earlier, maybe during TableHandle creation?

[the remark this answers is not in the document]
Comment (alexjo2144) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java:160]: From what I've tested with Databricks, generally the lowest required version is set in the protocol requirements based on the features the table is using. At least that was true for write features. 

Did you try making a table with name mapping set to none and see what the required version number is?

[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java:160]: It reports 2 as min reader version even if `delta.columnMapping.mode'='none'` or there's no such property when created the table with explicit `'delta.minReaderVersion'='2'` as far as I tried.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java [hunk 2]
```diff
@@ -171,6 +182,15 @@ public ConnectorPageSource createPageSource(
         return new DeltaLakePageSource(deltaLakeColumns, partitionKeys, pageSource.get(), split.getPath(), split.getFileSize(), split.getFileModifiedTime());
     }
 
+    private static ColumnMappingMode getColumnMappingMode(MetadataEntry metadata)
+    {
+        String columnMappingMode = metadata.getConfiguration().get(COLUMN_MAPPING_MODE_KEY);
+        if (columnMappingMode == null) {
+            return ColumnMappingMode.NONE;
+        }
+        return ColumnMappingMode.valueOf(columnMappingMode.toUpperCase(ENGLISH));
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java:191]: I intentionally chose "throw when a new mode is added", but the suggested way looks better. Let me fix it.

**Comment by findepi** [comment]
@ebyhr please rebase

@alexjo2144 PTAL

**Review by homar — APPROVED** [review]

**Review by alexjo2144 — APPROVED** [review]

**Review by findepi — APPROVED** [review]
be sure to squash

**Pull request merged**